namespace.go

```diff
@@ -153,6 +156,28 @@ func NamespaceWithEnvironmentBinding(name string) NamespaceOption {
 	}
 }
 
+// NamespaceWithAzureEnvironment sets the namespace's Environment, Suffix and ResourceURI parameters according
+// to the Azure Environment defined in "github.com/Azure/go-autorest/autorest/azure" package.
+// This allows to configure the library to be used in the different Azure clouds.
+// envName is the name of the cloud as defined in autorest : https://github.com/Azure/go-autorest/blob/b076c1437d051bf4c328db428b70f4fe22ad38b0/autorest/azure/environments.go#L34-L39
+func NamespaceWithAzureEnvironment(envName string) NamespaceOption {
+	return func(ns *Namespace) error {
+		azureEnv, err := azure.EnvironmentFromName(envName)
+		if err != nil {
+			return err
+		}
+		ns.Environment = azureEnv
+		ns.Suffix = azureEnv.ServiceBusEndpointSuffix
+		// TODO: when autorest includes ResourceIdentifier.ServiceBus
+		// ns.ServiceBusResourceURI = azureEnv.ResourceIdentifiers.ServiceBus
+		// if ns.ServiceBusResourceURI  == "" {
+		// 	 ns.ServiceBusResourceURI = serviceBusResourceURI
+		// }
```

Comment:
note that, after your answer in autorest repo, I see that this needs to wait for autorest to include the servicebus resource uri.